includes/Products/Feed.php

```diff
@@ -158,10 +158,14 @@ public function regenerate_feed() {
 	 */
 	public function schedule_feed_generation() {
 
-		$integration = facebook_for_woocommerce()->get_integration();
-
-		// only schedule if configured
-		if ( ! $integration || ! $integration->is_configured() || ! $integration->is_product_sync_enabled() ) {
+		$integration   = facebook_for_woocommerce()->get_integration();
+		$configured_ok = $integration && $integration->is_configured();
+
+		// Only schedule feed job if store has not opted out of product sync.
+		$store_allows_sync = $configured_ok && $integration->is_product_sync_enabled();
+		// Only schedule if has not opted out of feed generation (e.g. large stores).
+		$store_allows_feed = $configured_ok && $integration->is_legacy_feed_file_generation_enabled();
+		if ( ! $store_allows_sync || ! $store_allows_feed ) {
```

Comment:
```suggestion
		if ( ! ( $store_allows_sync && $store_allows_feed ) ) {
```
I think that this way we are more explicit about the intention.

Reply:
The suggested change would be clearer IMO if the if condition was the positive case, but since we have to negate that undermines the clarity. I'm leaving as is - the idea is that if either option is disabled, we'll bail - and the code reads like that. If that doesn't make sense let me know, we can revisit :)